tasks/geographica/uk/csv.py

```diff
@@ -0,0 +1,118 @@
+        for measure in measurements:
+            measure['geom_id'] = GEOGRAPHY_LEVELS[self.geography]
+        json_metadata = json.dumps(measurements)
+
+        result = session.execute(self._get_meta_query(json_metadata))
+        if result:
+            join_data = {}
+            join_data['numer'] = {}
+            if self.geography == 'GEO_PA':
+                colnames = ['geom.pa_id as geoid']
+            else:
+                colnames = ['geom.geographycode as geoid']
+
+            for data in result.fetchall():
+                join_data['numer'][data['numer_table']] = {'table': 'observatory.{}'.format(data['numer_table']),
+                                                           'join_column': data['numer_join_col']}
+                # All come from the same geometry tables so we use, by now, just one geometry
+                # TODO Make it possible to have multiple geometry tables
+                join_data['geom'] = {'table': 'observatory.{}'.format(data['geom_table']), 'join_column': data['geom_join_col']}
+                colnames.append(data['numer_col'])
+
+            measurement_result = session.execute(self._get_measurements_query(join_data, colnames))
+            if measurement_result:
+                measurements = measurement_result.fetchall()
+                self._generate_csv_file(colnames, measurements)
+            else:
+                LOGGER.error('No results for the queried measurements')
+
+        else:
+            LOGGER.error('No results for the defined measurements in the JSON file')
+
+    def _get_meta_query(self, metadata):
+        return '''SELECT meta->>'numer_tablename' numer_table, meta->>'numer_geomref_colname' numer_join_col,
+                         meta->>'numer_colname' numer_col, meta->>'geom_tablename' geom_table,
+                         meta->>'geom_geomref_colname' geom_join_col, meta->>'geom_colname' geom_col
+                  FROM json_array_elements(cdb_observatory.OBS_GetMeta(
+                       ST_MakeEnvelope(-179, 89, 179, -89, 4326), -- World bbox
+                       '{}'::json, 1, 1, 1)) meta
+            '''.format(metadata)
+
+    def _get_measurements_query(self, join_data, colnames):
+        joins = []
+        for join_table in join_data['numer'].values():
+            joins.append('LEFT JOIN {table} ON (geom.{geomcol} = {table}.{numercol})'.format(
+                table=join_table['table'], geomcol=join_data['geom']['join_column'],
+                numercol=join_table['join_column']
+            ))
+        return '''SELECT {cols}
+                  FROM {geom} geom {joins}
+               '''.format(cols=' ,'.join(colnames), geom=join_data['geom']['table'],
+                          joins=' '.join(joins))
+
+    def _generate_csv_file(self, headers, measurements):
+        try:
+            self.output().makedirs()
+            with(open(self.output().path, 'w+')) as csvfile:
+                headers[0]='geoid'
```

Comment:
Changing a header inside the function feels strange, doesn't it? Is there a good reason to do it here instead of outside?

Reply:
Yes, it's strange, but I haven't thought much about it because the CSV generation is in need of a refactor when we have all (more) the tasks defined